### PR TITLE
Added showOn options parameter function

### PR DESCRIPTION
**Pull request by mtgibbs**
We have conditionally shown controls on our site and didn't want to have to make individual tours for different user types.  Please take a look at this proposed update inspired by Issue 65.

https://github.com/HubSpot/shepherd/issues/65

JsFiddle showing it in action:

http://jsfiddle.net/mattgibbs/f8n0gLrL/4/

### CODE REVIEW

File: src/js/shepherd.js [hunk 1]
```diff
@@ -527,13 +528,19 @@ class Tour extends Evented {
     }
 
     if (next) {
-      this.trigger('show', {
-        step: next,
-        previous: this.currentStep
-      });
+      if (typeof next.options.showOn !== 'undefined' && !next.options.showOn()) {
+        const index = this.steps.indexOf(next);
+        var nextIndex = forward ? index + 1 : index - 1;
```
Comment (geekjuice) [inline src/js/shepherd.js]: One minor detail. `var` here should be `const`.

**Comment by geekjuice** [comment]
Other than the small comment above, I think you can go ahead and minor version bump this to `v1.2.0` since `showOn` is a new feature. To do that, just run `gulp version:minor && gulp build` to have all the `.json` files and distribution files updated.

Once that's done, I'll go ahead and merge this.

Thanks again for being patience while I was on hiatus :+1:

**Comment by mtgibbs** [comment]
Great!  Thanks for reviewing it.

I've committed your feedback, bumped the version, and committed the install output.

I really enjoy these tours and if I run into something else I'll send it your way.

**Comment by geekjuice** [comment]
:tada: :+1:

**Comment by dmackerman** [comment]
Is there no documentation for `showOn`?

**Comment by mtgibbs** [comment]
I guess not... :(

I didn't know where to submit the changes to the documentation.  But what is it you want to know about it?  Basically you have to define the showOn option as a function that returns a boolean.  If it is defined, when the step is stepped to it will execute the function and look at the return value to decide whether or not the step should be shown.  If it shouldn't be, it will skip to the next one.

**Pull request merged**